### ISSUE TITLE
ptvsd: enable_attach() should spawn adapter

### ISSUE THREAD

**Issue by int19h** [microsoft/ptvsd#1712]
In attach scenarios, `ptvsd.enable_attach()` is executed directly inside user code inside the process to be debugged, but there's no adapter at that point. Thus, the debug server needs to spawn the adapter in this case, and establish a communication channel with it.

**Comment by int19h** [comment]
In this model, when user calls `ptvsd.enable_attach((host, port))` in their code, what actually happens is:

1. ptvsd generates a cryptographically secure server access token (#1710).
1. ptvsd tells pydevd to open a DAP debug server socket on 127.0.0.1:0 using said token, and gets the actual port number for that.
1. ptvsd spawns adapter with a command line that tells it to connect to the debug server on the chosen port using the generated server token, and also to wait for incoming connections from IDE on host:port. 
1. adapter connects to pydevd on the chosen port, and sends the server token via "initialize" to authorize; no other messages are sent.
1. adapter waits for incoming connection from IDE.
1. IDE connects to adapter on host:port, starts sending messages.
1. adapter starts relaying messages to pydevd - debug session is fully active.

Note that pydevd doesn't need to be aware of any of this. It's the wrapper code in `ptvsd.enable_attach()` that does the adapter spawning, generates tokens, figures out which port is which etc.

@fabioz, one catch in this case is that the initial handshake between the adapter and the server via "initialize" would happen almost immediately, long before the IDE actually connects. In the current implementation, if pydevd is in background debug server mode (i.e. non-blocking `settrace()`), and it receives an incoming connection that _only_ does initialize, will it continue running user code? Or will it pause until "configurationDone"?

**Comment by int19h** [comment]
Note: this is not needed for `ptvsd.attach()`, because the adapter must already be running in that scenario. However, `ptvsd.attach()` will need to get a new argument to pass the _client_ access token through to `pydevd.settrace()`; and we'll need a corresponding CLI argument in `ptvsd.main`.

**Comment by fabioz** [comment]
@int19h pydevd should never start running user code before a `configurationDone` (if it does that it's a bug -- this could only happen in the previous design because `ptvsd` unblocked pydevd by using internal APIs, but with the new approach this should not be an issue).
